lib/ansible/modules/cloud/amazon/iam_cert.py

```diff
@@ -271,10 +260,12 @@ def main():
     cert_chain = module.params.get('cert_chain')
     dup_ok = module.params.get('dup_ok')
     if state == 'present':
-        cert = open(module.params.get('cert'), 'r').read().rstrip()
-        key = open(module.params.get('key'), 'r').read().rstrip()
-        if cert_chain is not None:
-            cert_chain = open(module.params.get('cert_chain'), 'r').read()
+        if module.params.get('cert') is not None:
+            cert = module.params.get('cert')
```

Comment:
This looks good! In addition to removing the ifs Can you add a required_if to the call to [AnsibleModule](https://github.com/ansible/ansible/pull/24206/files#diff-ee240e40955a7a90ec4e28effc4a4570L251)?
 ...something like required_if=[('state', 'present', ('cert', 'key', 'cert_chain'))] so the user gets a helpful error